dapr/proto/runtime/v1/dapr.proto

```diff
@@ -148,6 +153,37 @@ message GetStateRequest {
   map<string, string> metadata = 4;
 }
 
+// ComponentHealthResponseItem represents health of a single component item in the response
+message ComponentHealthResponseItem {
+  // The name of component
+  optional string component_name = 1;
+
+  // Type of component
+  optional string type = 2;
+
+  // Status of component, OK or NOT OK or UNDEFINED
+  optional string status = 3;
+
+  // error that is returned in case of health check failure
+  optional string error_code = 4;
+
+  // message that may provide more insight for error, if any
+  optional string message = 5;
+}
+
+// ComponentHealthResponse is the response that would comprise of HealthItems
+message ComponentHealthResponse {
+  repeated ComponentHealthResponseItem results = 1;
+}
+
+// CheckHealthRequest is the message to get health of component(s).
+// component_name is optional. If not specified, health of all components is returned.
+// If specified, health of the component with the specified name is returned.
+message ComponentHealthRequest {
+  // The name of component.
+  optional string component_name = 1;
```

Comment:
On the empty `message` and `error_code` etc, that's just when you encode them as JSON using the regular `json` package. Using `protojson` is recommended when encoding protos to JSON. But this shouldn't matter with gRPC since the data is encoded as protos and not JSON anyways.

Reply:
In grpc code here, I don't seem to be using json, I am just sending in the required items to response and rest is taken care of by self generated code  for proto - how can I make blank fields NOT to be printed in response, without optional  -  as far as I have understood, without optional, it might not understand difference between "" and nil